decoder/src/lib.rs

```diff
@@ -98,13 +98,69 @@ pub struct Table {
 /// Checks if the version encoded in the symbol table is compatible with this version of the
 /// `decoder` crate
 pub fn check_version(version: &str) -> Result<(), String> {
+    enum Kind {
+        // "1" or "0.1"
+        Semver,
+        // commit hash "e739d0ac703dfa629a159be329e8c62a1c3ed206"
+        Git,
+    }
+
+    impl Kind {
```

Comment:
question: why are these defined within the function instead of outside?

Reply:
the `enum` is only used within the function body so putting it inside avoids adding yet another type to the outer (`src/lib.rs`) type namespace.

Another way to keep the namespace tidy is use modules so this could have been written as: `mod version { enum Kind { .. } pub fn check(version: &str) -> Result<(), String> { .. } }`. We have not used modules in this crate at all so the file is over 1.6K lines ... we should eventually come back and refactor this into modules.

Reply:
alrighty, thank you for the clarification! +1 to the refactoring, but that's for another day ofc